cli/planner_parentDailyEmailSummary.php

```diff
@@ -0,0 +1,163 @@
+
+// Override the ini to keep this process alive
+ini_set('memory_limit', '2048M');
+ini_set('max_execution_time', 1800);
+set_time_limit(1800);
+
+// Prep for email sending later
+$mail = $container->get(Mailer::class);
+$mail->SMTPKeepAlive = true;
+                
+$sendReport = ['emailSent' => 0, 'emailFailed' => 0];
+
+$currentDate = date('Y-m-d');
+$gibbonSchoolYearID = $gibbon->session->get('gibbonSchoolYearID');
+
+$familyGateway = $container->get(FamilyGateway::class);
+$attendanceLogGateway = $container->get(AttendanceLogPersonGateway::class);
+$view = $container->get(View::class);
+
+$schoolLogCriteria = $attendanceLogGateway->newQueryCriteria()
+    ->sortBy('timestampTaken')
+    ->filterBy('notClass', true);
+
+$classLogCriteria = $attendanceLogGateway->newQueryCriteria()
+    ->sortBy(['timeStart', 'timeEnd', 'timestampTaken']);
+
+// Get all student data grouped by family
+$families = $familyGateway->selectFamiliesWithActiveStudents($gibbonSchoolYearID)->fetchGrouped();
+
+foreach ($families as $gibbonFamilyID => $students) {
+    // Get the adults in this family and filter by email settings
+    $familyAdults = $familyGateway->selectAdultsByFamily($gibbonFamilyID, true)->fetchAll();
+    $familyAdults = array_filter($familyAdults, function ($adult) {
+        return $adult['contactEmail'] == 'Y';
+    });
+
+    if (empty($familyAdults)) continue;
+    $content = '';
+
+    foreach ($students as $student) {
+        // Get school-wide attendance logs
+        $logs = $attendanceLogGateway->queryByPersonAndDate($schoolLogCriteria, $student['gibbonPersonID'], $currentDate);
+        $schoolLog = $logs->getRow(count($logs) - 1);
+
+        // Get class attendance logs
+        $classLogs = $attendanceLogGateway->queryClassAttendanceByPersonAndDate($classLogCriteria, $gibbonSchoolYearID, $student['gibbonPersonID'], $currentDate);
+        $classLogs = array_filter($classLogs->toArray(), function ($log) {
+            return !empty($log['gibbonAttendanceLogPersonID']);
+        });
+
+        // Format the student attendance log for emailing
+        $content .= $view->fetchFromTemplate('cli/attendanceEmailParent.twig.html', [
+            'student' => $student,
+            'schoolLog' => $schoolLog,
+            'classLogs' => $classLogs,
+        ]);
+    }
+
+    // Format the email subject and content
+    $subject = __('Daily Attendance Summary for {context}', ['context' => Format::date(date('Y-m-d'))]);
+
+    $body = $parentDailyEmailSummaryIntroduction;
+    $body .= '<br/><br/>';
+    $body .= $content;
+    $body .= '<br/>';
+    $body .= $parentDailyEmailSummaryPostScript;
+
+    // Add recipients and sender
+    foreach ($familyAdults as $adult) {
+        $mail->AddAddress($adult['email'], Format::name('', $adult['preferredName'], $adult['surname'], 'Parent', false, true));
+    }
+    $mail->setDefaultSender($subject);
```

Comment:
Adding the $subject as the default sender?

Reply:
Aha right, in hindsight that method signature doesn't look quite right with the method name. The default sender is already in session data, so it's setting that, then prepending the organisationNameShort to the beginning of the subject:

```
public function setDefaultSender($subject)
    {
        $this->Subject = $this->session->get('organisationNameShort').' - '.$subject;
        $this->SetFrom($this->session->get('organisationEmail'), $this->session->get('organisationName'));
    }
```

Reply:
Right, makes sense! Thanks.